src/utils/mesh_utils.ts

```diff
@@ -23,20 +23,25 @@ export const meshUtils = {
     orderInfoToAPIOrder: (
         orderEvent: OrderEvent | AcceptedOrderInfo | RejectedOrderInfo | OrderInfo,
     ): APIOrderWithMetaData => {
-        const remainingFillableTakerAssetAmount = (orderEvent as OrderEvent).fillableTakerAssetAmount
-            ? (orderEvent as OrderEvent).fillableTakerAssetAmount
-            : ZERO;
-        return {
+        const { orderHash, fillableTakerAssetAmount, signedOrder } = orderEvent as OrderEvent;
+        const apiOrder: APIOrderWithMetaData = {
             // orderEvent.signedOrder comes from mesh with string fields, needs to be serialized into SignedOrder
-            order: orderUtils.deserializeOrder(orderEvent.signedOrder as any), // tslint:disable-line:no-unnecessary-type-assertion
+            order: orderUtils.deserializeOrder(signedOrder as any), // tslint:disable-line:no-unnecessary-type-assertion
             metaData: {
-                orderHash: orderEvent.orderHash,
-                remainingFillableTakerAssetAmount,
-                state:
-                    (orderEvent as OrderEvent).endState ||
-                    meshUtils.rejectedCodeToOrderState((orderEvent as RejectedOrderInfo).status.code),
+                orderHash,
+                remainingFillableTakerAssetAmount: fillableTakerAssetAmount || ZERO,
             },
         };
+        // populate order state
+        apiOrder.metaData.state = (orderEvent as OrderEvent).endState;
+        if (apiOrder.metaData.state === undefined) {
+            const r = orderEvent as RejectedOrderInfo;
+            apiOrder.metaData.state =
+                r.status && r.status.code
+                    ? meshUtils.rejectedCodeToOrderState(r.status.code)
+                    : OrderEventEndState.Invalid;
+        }
+        return apiOrder;
```

Comment:
This is the new logic. Previously would sometimes return undefined and then throw an error when saving to db.